analyticsclient/tests/test_engagement_timeline.py

```diff
@@ -0,0 +1,57 @@
+import json
+import re
+
+import httpretty
+
+from analyticsclient.exceptions import ClientError
+from analyticsclient.tests import ClientTestCase
+
+
+class EngagementTimelineTests(ClientTestCase):
+
+    def setUp(self):
+        super(EngagementTimelineTests, self).setUp()
+        self.username = 'edx'
+        self.course_id = 'edX/DemoX/Demo_Course'
+        self.engagement_timeline = self.client.engagement_timeline(self.username, self.course_id)
+        httpretty.enable()
+
+    def tearDown(self):
+        super(EngagementTimelineTests, self).tearDown()
+        httpretty.disable()
+
+    def test_not_found(self):
+        not_a_valid_course_id = 'foobar'
+        uri = self.get_api_url('engagement_timelines/{username}/?course_id={course_id}'
+                               .format(username=self.username, course_id=not_a_valid_course_id))
+        uri = re.compile(r'^' + re.escape(uri) + r'.*$')
+        httpretty.register_uri(httpretty.GET, uri, status=404)
+
+        engagement_timeline = self.client.engagement_timeline(self.username, not_a_valid_course_id)
+        self.assertRaises(ClientError, engagement_timeline.get)
+
+    def test_invalid_parameter(self):
+        uri = self.get_api_url('engagement_timelines/{username}/'.format(username=self.username))
+        uri = re.compile(r'^' + re.escape(uri) + r'.*$')
+        httpretty.register_uri(httpretty.GET, uri, status=400)
+
+        engagement_timeline = self.client.engagement_timeline(self.username, self.course_id)
+        self.assertRaises(ClientError, engagement_timeline.get)
+
+    @httpretty.activate
```

Comment:
Is this decorator needed since you have `httpretty.enable()` in the `setUp` and `httpretty.disable()` in the `tearDown`?

Reply:
heh probably not :-)